.dotnet/src/Custom/FineTuning/FineTuningJobOperation.Protocol.cs

```diff
@@ -44,122 +53,84 @@ internal FineTuningJobOperation(
-    /// request. </param>
-    /// <returns> The rehydrated operation. </returns>
+    /// <param name="client"> The <see cref="FineTuningClient"/> used to obtain the operation status from the service. </param>
+    /// <param name="rehydrationToken"> The rehydration token corresponding to the operation to rehydrate. </param>
+    /// <param name="options"> The request options, which can override default behaviors of the client pipeline on a per-call basis. </param>
+    /// <returns> The rehydrated operation <see cref="FineTuningJobOperation"/>. </returns>
     /// <exception cref="ArgumentNullException"> <paramref name="client"/> or <paramref name="rehydrationToken"/> is null. </exception>
-    public static async Task<FineTuningJobOperation> RehydrateAsync(FineTuningClient client, ContinuationToken rehydrationToken, CancellationToken cancellationToken = default)
+    public static FineTuningJobOperation Rehydrate(FineTuningClient client, ContinuationToken rehydrationToken, RequestOptions options)
     {
         Argument.AssertNotNull(client, nameof(client));
         Argument.AssertNotNull(rehydrationToken, nameof(rehydrationToken));
 
         FineTuningJobOperationToken token = FineTuningJobOperationToken.FromToken(rehydrationToken);
 
-        ClientResult result = await client.GetJobAsync(token.JobId, cancellationToken.ToRequestOptions()).ConfigureAwait(false);
+        ClientResult result = client.GetJob(token.JobId, options);
         PipelineResponse response = result.GetRawResponse();
 
-        using JsonDocument doc = JsonDocument.Parse(response.Content);
-        string status = doc.RootElement.GetProperty("status"u8).GetString()!;
-
-        return client.CreateCreateJobOperation(token.JobId, status, response);
+        return client.CreateCreateJobOperation(token.JobId, response);
     }
 
-    /// <summary>
-    /// Recreates a <see cref="FineTuningJobOperation"/> from a rehydration token.
-    /// </summary>
-    /// <param name="client"> The <see cref="FineTuningClient"/> used to obtain the 
-    /// operation status from the service. </param>
-    /// <param name="rehydrationToken"> The rehydration token corresponding to 
-    /// the operation to rehydrate. </param>
-    /// <param name="cancellationToken"> A token that can be used to cancel the 
-    /// request. </param>
-    /// <returns> The rehydrated operation. </returns>
-    /// <exception cref="ArgumentNullException"> <paramref name="client"/> or <paramref name="rehydrationToken"/> is null. </exception>
-    public static FineTuningJobOperation Rehydrate(FineTuningClient client, ContinuationToken rehydrationToken, CancellationToken cancellationToken = default)
+    /// <inheritdoc cref="Rehydrate(FineTuningClient, ContinuationToken, RequestOptions)"/>
+    public static async Task<FineTuningJobOperation> RehydrateAsync(FineTuningClient client, ContinuationToken rehydrationToken, RequestOptions options)
```

Comment:
I think it might be better to keep this as a specified CancellationToken so people know that is what it is used for. Theoretically the actual request options if any are kept in the ContinuationToken right?

Reply:
I think CancellationToken shouldn't be used in Protocol layer. There's a copy of this method in the convenience layer with cancellation token